File: TeachersRating.API/Entities/Department.cs
namespace TeachersRating.API.Entities;

public class Department : BaseEntity
{
    public Guid InstituteId { get; set; }
    public Institute? Institute { get; set; } = null;

    public string LongName { get; set; } = string.Empty;
    public string ShortName { get; set; } = string.Empty;

    public string DepartmentUrl { get; set; } = string.Empty;
    public string WorkersPageUrl { get; set; } = string.Empty;

    public ICollection<Worker> Workers { get; set; } = new List<Worker>();
}

File: NulpScrapper/models/Department.ts
import type { Institute } from "./Institute.js";

export interface Department {
    institute: Institute;
    href: string;
    longName: string;
    shortName: string;
    workersPageHref: string;
}

export function createDepartment(institute: Institute, href: string, fullName: string, shortName: string, workersPageHref: string): Department {
    return { institute, href, longName: fullName, shortName, workersPageHref } as Department;
}